### PR TITLE
Dat 3804 clicking on infobox preview background exits edit mode on currently selected item

### PR DESCRIPTION
**Pull request by RafLeszczynski**
## Links
* https://wikia-inc.atlassian.net/browse/DAT-3804

## Description

New interaction added. When user is editing currently selected item inside infobox builder, clicking on infobox preview background deselects item and exits edit mode

## Reviewers

@rogatty @SebastianMarzjan @dianafa

### CODE REVIEW

File: front/main/tests/unit/controllers/infobox-builder-test.js [hunk 1]
```diff
@@ -1,32 +1,81 @@
...
+//				message: 'should not reset item in edit mode if it match actionTrigger'
+//			}
+//		];
+//
+//	controller.set('model', modelMock);
+//	cases.forEach((testCase) => {
+//		controller.send('handleItemInEditMode', testCase.actionTrigger);
+//		assert.equal(resetEditModeSpy.called, testCase.wasReset, testCase.message);
+//	});
+//});
```
Comment (dianafa) [inline front/main/tests/unit/controllers/infobox-builder-test.js]: what happened here?

File: front/main/tests/integration/components/infobox-builder-test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+	this.set('setEditItem', setEditItemSpy);
+	this.render(hbs`{{infobox-builder setEditItem=setEditItem}}`);
+
+	this.$(previewSelector).click();
+
+	assert.equal(setEditItemSpy.called, true);
+	assert.equal(setEditItemSpy.calledWith(null), true);
+});
+
```
Comment (rogatty) [inline front/main/tests/integration/components/infobox-builder-test.js]: Too much, it's too much :see_no_evil:

File: front/main/tests/unit/components/infobox-builder-test.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
+import {test, moduleForComponent} from 'ember-qunit';
+
+moduleForComponent('infobox-builder', 'Unit | Component | on hover tooltip', {
```
Comment (rogatty) [inline front/main/tests/unit/components/infobox-builder-test.js]: It's not a tooltip.

File: front/main/tests/unit/components/infobox-builder-test.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
+import {test, moduleForComponent} from 'ember-qunit';
+import sinon from 'sinon';
+
+moduleForComponent('infobox-builder', 'Unit | Component | on hover tooltip', {
```
Comment (rogatty) [inline front/main/tests/unit/components/infobox-builder-test.js]: It's not a tooltip.

**Comment by rogatty** [comment]
Two minor things and it'll LGTM.

File: front/main/tests/integration/components/infobox-builder-test.js (hunk 2 above)
Comment (RafLeszczynski) [inline front/main/tests/integration/components/infobox-builder-test.js]: could you be more specific ?

Comment (rogatty) [inline front/main/tests/integration/components/infobox-builder-test.js]: There are two `\n`s here and that's why there is an empty line visible on Github. One would be the perfect number of them.

**Pull request merged**